src/main/java/prefs/Preferences.java

```
@@ -11,122 +17,262 @@
+     * @param configDirectory The directory that the config file is held in.
+     * @param configFileName The filename of the config file on disk
+     * @param configClass The class of the config to load
+     * @return The loaded Config instance
      */
-    public static Preferences create(String configFileName) {
-        return new Preferences(configFileName, true);
-    }
+    private <T extends Config> T loadConfig(String configDirectory, String configFileName, Class<T> configClass) {
+        FileHelper configFileHelper = new FileHelper(configDirectory, configFileName);
+        String fileContents;
+        if (!configFileHelper.exists()) {
+            fileContents = DEFAULT_FILE_CONTENTS;
+        } else {
+            try {
+                fileContents = configFileHelper.loadFileContents();
+            } catch (IOException e) {
+                // if we can't read the file, just ignore the existing file
+                HTLog.error(logger, e);
+                logger.error(e.toString());
+                fileContents = DEFAULT_FILE_CONTENTS;
+            }
+        }
 
-    public void saveGlobalConfig() {
-        fileHandler.saveGlobalConfig(global);
+        return createConfig(fileContents, configClass);
     }
 
-    private void initGlobalConfig() {
-        global = fileHandler.initGlobalConfig();
+    /**
+     * Saves the session and user configs to file
+     */
+    private void save() {
+        try {
+            saveConfig(sessionConfig, configDirectory, sessionConfigFileName, SessionConfig.class);
+
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+            logger.error("Could not save session config");
+        }
+
+        try {
+            saveConfig(userConfig, configDirectory, userConfigFileName, UserConfig.class);
+
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+            logger.error("Could not save user config");
+        }
```

Swallowing exceptions like this may be a problem.